packages/next/client/link.tsx

```diff
@@ -135,6 +135,11 @@ class Link extends Component<LinkProps> {
       this.props.href,
       this.props.as
     )
+    if (parsedHref == null) {
```

Comment:
This needs to be wrapped in `if(typeof process.env.NODE_ENV !== 'production')` 👍